recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+from conans.errors import ConanInvalidConfiguration
+from conans import ConanFile, Meson, tools
+
+required_conan_version = ">=1.29.0"
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    homepage = "https://github.com/libsigcplusplus/libsigcplusplus"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    description = (
+        "libsigc++ implements a typesafe callback system for standard C++."
+    )
+    topics = ("conan", "libsigcpp", "callback")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("meson/0.56.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            self.options.shared = True
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if (
+            not self.options.shared
```

Comment:
The reason a static build on MSVC fails, is because meson is dumb enough to create a `.a` static library instead of a `.lib` on MSVC.
See https://github.com/conan-io/conan-center-index/pull/3427#issuecomment-723074886
```
Installing sigc++\libsigc-2.0.a to C:\J\w\cci_PR-3427@2\.conan\data\libsigcpp\2.10.4\_\_\package\63da998e3642b50bee33f4449826b2d623661505\lib
```

In `package()`, you need to add something like the following:
```python
if self.settings.compiler and not self.options.shared:
    os.rename(os.path.join(self.package_folder, "lib", "libsigc-2.0.a"),
              os.path.join(self.package_folder, "lib", "sigc-2.0.lib"))
```

Reply:
Are you sure? I tested the libsigcpp release directly with meson (no intermediate conan whatsoever) and then Meson seems to generate the right files right away. So I am inclined it has something to do with the combo Conan + Meson (although I must admit I did not look any further into it).

Reply:
Check the logs of MSVC2015 I referenced to above.

Reply:
Indeed:

```
Installing sigc++\libsigc-2.0.a to C:\J\w\cci_PR-3427@2\.conan\data\libsigcpp\2.10.4\_\_\package\63da998e3642b50bee33f4449826b2d623661505\lib
```

Reply:
@bverhagen 
I guess this throw can now be removed?

Reply:
Pango PR has a similar problem, same solution, rename the library: https://github.com/conan-io/conan-center-index/pull/3459